legal-api/src/legal_api/reports/report.py

```diff
@@ -128,18 +128,27 @@ def _get_environment():
     def _get_template_filename(self):
         return '{}.html'.format(self._filing.filing_type)
 
-    def _get_template_data(self):
+    def _get_template_data(self):  # pylint: disable=too-many-branches
         filing = copy.deepcopy(self._filing.filing_json['filing'])
 
         # set registered office address from either the COA filing or status quo data in AR filing
         try:
             if filing.get('changeOfAddress'):
-                filing['registeredOfficeAddress'] = filing['changeOfAddress']
+                if filing.get('changeOfAddress').get('offices'):
+                    filing['registeredOfficeAddress'] = filing['changeOfAddress']['offices']['registeredOffice']
+                else:
+                    filing['registeredOfficeAddress'] = filing['changeOfAddress']
             else:
-                filing['registeredOfficeAddress'] = {
-                    'deliveryAddress': filing['annualReport']['deliveryAddress'],
-                    'mailingAddress': filing['annualReport']['mailingAddress']
-                }
+                if filing.get('annualReport', {}).get('deliveryAddress'):
+                    filing['registeredOfficeAddress'] = {
+                        'deliveryAddress': filing['annualReport']['deliveryAddress'],
+                        'mailingAddress': filing['annualReport']['mailingAddress']
+                    }
+                else:
+                    filing['registeredOfficeAddress'] = {
+                        'deliveryAddress': filing['annualReport']['offices']['registeredOffice']['deliveryAddress'],
+                        'mailingAddress': filing['annualReport']['offices']['registeredOffice']['mailingAddress']
+                    }
```

Comment:
Does the same thing need to be done for Records Addresses?

Reply:
Not at this point

Reply:
it will be done once the ticket is picked up for outputs, which should be soon since the designs are now done